src/item_loader.rs

```diff
@@ -114,16 +114,18 @@ impl ItemLoader for TomlItemLoader {
     }
 }
 
-fn load_items_from_entries(entries: impl Iterator<Item = DirEntry>) -> HashMap<String, Item> {
-    entries
+fn load_items_from_entries(entries: impl Iterator<Item = DirEntry>) -> Result<HashMap<String, Item>, Box<dyn Error>> {
+    let items = entries
         .map(|entry| entry.path().to_string_lossy().into_owned())
         .filter(|file_path| file_path.ends_with(".toml"))
-        .map(|file_path| load_items_from_file(&file_path).into_iter())
-        .flatten()
+        .map(|file_path| load_items_from_file(&file_path))
+        .collect::<Result<Vec<_>, Box<dyn Error>>>()?
```

Comment:
Does type inference work on the Error type here?

Reply:
60b2d5e